packages/access-client/src/agent-use-cases.js

```diff
@@ -170,6 +189,43 @@ export async function waitForDelegationOnSocket(access, opts) {
   throw new TypeError('Failed to get delegation')
 }
 
+/**
+ * @typedef {{signal?: AbortSignal }} AuthorizationWaiterOpts
+ * @typedef {(accessAgent: AccessAgent, opts: AuthorizationWaiterOpts) => Promise<Iterable<Ucanto.Delegation>> } AuthorizationWaiter
+ */
+
+/**
+ * Wait for the authorization process to complete by waiting on a
+ * well-known websocket endpoint for the access-api server to
+ * receive and forward a session delegation from the authorization
+ * email flow.
+ *
+ * @type AuthorizationWaiter
+ */
+export async function waitForAuthorizationOnSocket(access, opts = {}) {
+  const delegation = await waitForDelegationOnSocket(access, opts)
```

Comment:
If we're going to make this a little backwards incompatible, can we just change the `waitForDelegationOnSocket` to match type `AuthorizationWaiter`? and avoid exporting/maintaining both of `waitForDelegationOnSocket` and `waitForAuthorizationOnSocket` with such similar types?

Reply:
yes, though since waitForDelegationOnSocket is definitely in use in a couple places I'd definitely want to do a major version release in this case

Reply:
tbh I think I'd feel better `@deprecated`ing `waitForDelegationOnSocket` for now and removing it after the next major version release, wdyt?

Reply:
ok added @deprecated